### DISCUSSION TITLE
laravel mysql database sharding database sharding table

### DISCUSSION THREAD

**Discussion by jasonccs** [laravel/framework#48499] [Q&A]
### Laravel Version

10.23.1

### PHP Version

8.2.3

### Database Driver & Version

5.7

### Description

We are currently looking solution/library for Sharding our DB with Laravel. Do you know the good one?

### Steps To Reproduce

We are currently looking solution/library for Sharding our DB with Laravel. Do you know the good one?

**Comment by jasonccs** [comment]
Does Laravel have a plugin package with MySQL database ? My business data volume is very large, so I am considering dividing it into databases and tables to handle queries on large data, in order to improve efficiency

If I have 4 tables in the same MySQL database and the table structure is the same, as shown in Figure users01,users02,users03,users04,

    public function up(): void
    {
        Schema::create('users', function (Blueprint $table) {
            $table->id();
            $table->string('name');
            $table->string('email')->unique();
            $table->timestamp('email_verified_at')->nullable();
            $table->string('password');
            $table->rememberToken();
            $table->timestamps();
        });
    }
    

Can I perform data sharding according to the specified fields, add, delete, modify, and query data, and perform related operations on the associated table,

**Comment by wizzymore** [comment]
Creating tables like: table_nameXX is a very bad idea.
I think the best approach for you will be partitioning the tables.

https://dev.mysql.com/doc/refman/8.0/en/partitioning.html
